src/Ads/AssetSuggestionsService.php

```diff
@@ -134,20 +139,22 @@ public function get_assets_suggestions( int $id, string $type ): array {
 	/**
 	 * Get URL for a specific post or term.
 	 *
-	 * @param int    $id Post or Term ID.
-	 * @param string $type Only possible values are post or term.
+	 * @param int    $id Post ID, Term ID or self::HOMEPAGE_KEY_ID
+	 * @param string $type Only possible values are post, term and homepage.
 	 *
 	 * @return string The URL.
 	 * @throws Exception If the ID is invalid.
 	 */
 	protected function get_url( int $id, string $type ): string {
 		if ( $type === 'post' ) {
 			$url = get_permalink( $id );
-		} else {
+		} elseif ( $type === 'term' ) {
 			$url = get_term_link( $id );
+		} else {
+			$url = get_bloginfo( 'url' );
 		}
 
-		if ( $url === false || is_wp_error( $url ) ) {
+		if ( $url === false || is_wp_error( $url ) || empty( $url ) ) {
```

Comment:
💅  since [empty()](https://www.php.net/manual/en/function.empty.php) checks also if $url is falsy we can consider to  omit `$url === false`